File: src/main/java/org/radarcns/config/SourceStatisticsStreamConfig.java
package org.radarcns.config;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import org.radarcns.stream.statistics.SourceStatisticsStream;

public class SourceStatisticsStreamConfig extends SingleStreamConfig {
    private String name;

    private List<String> topics;

    @JsonProperty("output_topic")
    private String outputTopic = "source_statistics";

    @JsonProperty("max_batch_size")
    private int maxBatchSize = 1000;

    @JsonProperty("flush_timeout")
    private long flushTimeout = 60_000L;

    public SourceStatisticsStreamConfig() {
        setStreamClass(SourceStatisticsStream.class);
    }

    public List<String> getTopics() {
        return topics;
    }

    public void setTopics(List<String> topics) {
        this.topics = topics;
    }

    public String getOutputTopic() {
        return outputTopic;
    }

    public void setOutputTopic(String outputTopic) {
        this.outputTopic = outputTopic;
    }

    public int getMaxBatchSize() {
        return maxBatchSize;
    }

    public void setMaxBatchSize(int maxBatchSize) {
        this.maxBatchSize = maxBatchSize;
    }

    public long getFlushTimeout() {
        return flushTimeout;
    }

    public void setFlushTimeout(long flushTimeout) {
        this.flushTimeout = flushTimeout;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/kotlin/org/radarbase/config/SourceStatisticsStreamConfig.kt
package org.radarbase.config

import com.fasterxml.jackson.annotation.JsonProperty
import org.radarbase.stream.statistics.SourceStatisticsStream

data class SourceStatisticsStreamConfig(
    var name: String? = null,

    var topics: List<String>? = null,

    @param:JsonProperty("output_topic")
    var outputTopic: String = "source_statistics",

    @param:JsonProperty("max_batch_size")
    var maxBatchSize: Int = 1000,

    @param:JsonProperty("flush_timeout")
    var flushTimeout: Long = 60_000L,
) : SingleStreamConfig() {
    init {
        streamClass = SourceStatisticsStream::class.java
    }
}
